### PR TITLE
Improve repo sync script

### PR DESCRIPTION
**Pull request by SuperQ**
* Change the Makefile.common update script to be more generic.
* Sync CODE_OF_CONDUCT.md and LICENSE files.
* Rename code-of-conduct.md to match github naming convention[0]

[0]: https://help.github.com/en/github/building-a-strong-community/adding-a-code-of-conduct-to-your-project

Signed-off-by: Ben Kochie <superq@gmail.com>

### CODE REVIEW

File: scripts/sync_repo_files.sh [hunk 1]
```diff
@@ -7,10 +7,10 @@ set -uo pipefail
...
 git_mail="prometheus-team@googlegroups.com"
 git_user="prombot"
-branch="makefile_common"
-commit_msg="makefile: update Makefile.common with newer version"
-pr_title="Synchronize Makefile.common from prometheus/prometheus"
-pr_msg="Propagating changes from master Makefile.common located in prometheus/prometheus."
+branch="repo_sync"
+commit_msg="Update common Prometheus files"
+pr_title="Synchronize common files from prometheus/prometheus"
+pr_msg="Propagating changes from prometheus/prometheus master."
```
Comment (paulfantom) [inline scripts/sync_repo_files.sh:13]:
```suggestion
pr_msg="Propagating changes from prometheus/prometheus default branch."
```

Just a nit, but I think it sounds a bit better and is future proof :)

File: scripts/sync_repo_files.sh [hunk 2]
```diff
@@ -19,11 +19,13 @@ if [ -z "${GITHUB_TOKEN}" ]; then
   exit 1
 fi
 
+# List of files that should be synced.
+SYNC_FILES="CODE_OF_CONDUCT.md LICENSE Makefile.common"
```
Comment (paulfantom) [inline scripts/sync_repo_files.sh:23]: LICENSE usually have a different year in it and cannot be simply copied.

Comment (paulfantom) [reply scripts/sync_repo_files.sh:23]: Oh, prometheus don't fill [those fields](https://github.com/prometheus/prometheus/blob/master/LICENSE#L189) so disregard my previous comment.

**Comment by SuperQ** [comment]
Ugh, stupid DCO sign-off and suggestions.

**Pull request merged**